### ISSUE TITLE
Adding addtional JS when Particle is part of drupal

### ISSUE THREAD

**Issue by danny-englander** [phase2/particle#472]
Hi, I can't seem to find any documentation on adding additional JS when particle is at the root of a drupal theme. I tried this in `/web/themes/custom/particle/source/default/_meta/_01-foot.twig`

```html
<script src="{{ paths.assets }}/uikit.min.js"></script>
```
... and the script is located at: `web/themes/custom/particle/apps/drupal`

I run `npm start` and the script does show up in Pattern Lab in web inspector as:

`<script src="../../../assets/uikit.min.js"></script>`

However that's just giving a 404 in Pattern Lab

```bash
http://0.0.0.0:8080/app-pl/assets/uikit.min.js net::ERR_ABORTED 404 (Not Found)
```
I also added the script as a library in `particle.libraries.yml` so I am not sure what I am doing wrong.

**Comment by tjheffner** [comment]
if it's a standalone script that's needed in both PL & drupal you'll probably want to add new entry points for webpack. you can see an example of this in the default drupal app for jquery (`apps/drupal/drupal-jquery.js` and `apps/drupal/webpack.config.js`). the entry point will ensure webpack bundles & moves it to the expected dist/assets directory. you should add this entry point for the PL app also.

if this script just needs to exist in drupal (and not in pattern lab), you don't necessarily need the extra entry point, but you will need to [add your library to the page](https://www.drupal.org/docs/8/creating-custom-modules/adding-stylesheets-css-and-javascript-js-to-a-drupal-8-module), either with preprocessing or by creating/editing a html.html.twig template.

**Comment by danny-englander** [comment]
@tjheffner Thanks for the info, that makes sense. I tried adding:

```javascript
  'uikit.min': [path.resolve(__dirname, 'uikit.min.js')],
```
... to just under`'drupal-jquery` in `apps/drupal/webpack.config.js` and then I ran `npm run build:drupal` but am getting an error with regard to my theme name which I've replaced everywhere. Since I've done that, I think I'll need to create a reduced test case using particle as is out of the box so I'm sure it's nothing to do with my custom theme name. I'll post back if I can't get it working with a fresh install.
